prometheus/promhttp/delegator_test.go

```diff
@@ -0,0 +1,28 @@
+// Copyright 2024 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package promhttp
+
+import (
+	"net/http/httptest"
+	"testing"
+)
+
+func TestResponseWriterDelegatorUnwrap(t *testing.T) {
```

Comment:
This test is a bit shallow it only tests single variable return 🙈 We could test if the HTTP lib will call Unwrap to get to those Flush methods properly. This would test our method from two sides:
* If the correct thing is passed and returned
* If our signature of the method actually works with http net lib.

What do you think? 🤔

Reply:
Sure, I've implemented a custom ResponseWriter and now it checks whether the methods were propagated correctly by http.ResponseController

I've omitted a test for Flush method, because the library seems to try FlushError instead: https://cs.opensource.google/go/go/+/refs/tags/go1.21.4:src/net/http/responsecontroller.go;l=51, to actually return an error